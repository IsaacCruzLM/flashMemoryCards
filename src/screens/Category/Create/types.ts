import {DefaultFormProps} from '../../../components/Form/types';

export interface CreateFormProps extends DefaultFormProps {
  values: formValues;
}

export interface formValues {
  name: string;
  icon: string;
  color: string;
}
