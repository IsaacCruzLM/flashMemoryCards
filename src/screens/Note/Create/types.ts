import {DefaultFormProps} from '../../../components/Form/types';

export interface CreateProps extends DefaultFormProps {}
