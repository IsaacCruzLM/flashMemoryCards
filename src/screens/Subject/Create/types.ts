import {DefaultFormProps} from '../../../components/Form/types';
import {CategoryModelType} from '../../../databases/models/categoryModel';

export interface CreateProps extends DefaultFormProps {
  route: object;
  category: CategoryModelType;
}

export interface CreateFormProps extends DefaultFormProps {
  values: formValues;
}

export interface formValues {
  name: string;
  icon: string;
  color: string;
}
