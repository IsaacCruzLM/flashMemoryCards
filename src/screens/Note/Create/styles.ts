import {StyleSheet} from 'react-native';
// import themes from '../../../styles/themes';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: '100%',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  formContainer: {
    width: '100%',
  },
});

export default styles;
