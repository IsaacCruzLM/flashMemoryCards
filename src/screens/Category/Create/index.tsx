import React from 'react';
import {View} from 'react-native';

import NavigationService from '../../../navigation/NavigationService';
import insertItenInWMDB from '../../../databases/utils/insertItenInWMDB';

import DefaultContainerView from '../../../components/DefaultContainerView';
import TextInput from '../../../components/TextInput';
import Form from '../../../components/Form';
import SelectIcon from '../../../components/SelectIcon';
import ColorPicker from '../../../components/ColorPicker';
import Button from '../../../components/Button';

import styles from './styles';
import {CreateFormProps, formValues} from './types';

const Create = () => {
  const cancelAction = (resetForm: () => void) => {
    resetForm();
    NavigationService.navigate('Categories');
  };

  const createAction = async (values: formValues | Object) => {
    await insertItenInWMDB('categories', {...values, createdAt: new Date()});
    NavigationService.navigate('Categories');
  };

  return (
    <DefaultContainerView>
      <Form
        form={({
          handleChange,
          handleBlur,
          setFieldValue,
          resetForm,
          handleSubmit,
          values,
        }: CreateFormProps) => (
          <View style={styles.formContainer}>
            <TextInput
              label={'Nome da categoria'}
              setText={handleChange('name')}
              onBlur={handleBlur('name')}
              placeholder={'Nome da categoria'}
              value={values.name}
            />
            <SelectIcon
              onPress={iconLabel => setFieldValue('icon', iconLabel)}
            />
            <ColorPicker
              iconName={values.icon}
              onChangeColor={color => setFieldValue('color', color)}
            />
            <Button
              label="Criar Categoria"
              onPress={handleSubmit}
              style={styles.createButton}
            />
            <Button
              label="Cancelar"
              modeParam="outlined"
              onPress={() => cancelAction(resetForm)}
            />
          </View>
        )}
        initialValues={{name: '', icon: '', color: ''}}
        onSubmit={values => createAction(values)}
      />
    </DefaultContainerView>
  );
};

export default Create;
