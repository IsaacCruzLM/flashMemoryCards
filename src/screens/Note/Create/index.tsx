import React, {useEffect, useState} from 'react';
import {Keyboard, Text, View} from 'react-native';
import withObservables from '@nozbe/with-observables';
import {withDatabase} from '@nozbe/watermelondb/DatabaseProvider';
import {compose} from 'recompose';
import get from 'lodash/get';
import cloneDeep from 'lodash/cloneDeep';

import DefaultContainerView from '../../../components/DefaultContainerView';
import Form from '../../../components/Form';
import Select from '../../../components/Select';
import TextEditor from '../../../components/TextEditor';
import SelectMultiple from '../../../components/SelectMultiple';
import Button from '../../../components/Button';
import TextInput from '../../../components/TextInput';

import WmdbUtils from '../../../databases/utils';

import styles from './styles';
import {CreateFormProps, CreateProps, formValues} from './types';

const Create: React.FunctionComponent<CreateProps | any> = ({
  categories,
  subjects,
  route,
}) => {
  const [step, setStep] = useState(1);
  const [keyboardStatus, setKeyboardStatus] = useState('');

  useEffect(() => {
    const showSubscription = Keyboard.addListener('keyboardDidShow', () => {
      setKeyboardStatus('Keyboard Shown');
    });
    const hideSubscription = Keyboard.addListener('keyboardDidHide', () => {
      setKeyboardStatus('Keyboard Hidden');
    });

    return () => {
      showSubscription.remove();
      hideSubscription.remove();
    };
  }, []);

  const translateOptions = (optionsArray: any) =>
    optionsArray.map(({id, name, icon, color}: any) => ({
      label: name,
      value: id,
      iconName: icon,
      iconColor: color,
    }));

  const submitAction = async (values: formValues | Object) => {
    const newDataObject = cloneDeep(values);
    const relationships = get(newDataObject, 'subjects', []).forEach(id => ({
      subject: id,
    }));
    delete newDataObject.subjects;

    console.log("Flag", newDataObject, relationships);
  };

  return (
    <DefaultContainerView>
      <Form
        form={({
          setFieldValue,
          handleSubmit,
          handleChange,
          handleBlur,
          values,
        }: CreateFormProps) => (
          <View style={styles.formContainer}>
            <View>
              {step === 1 ? (
                <>
                  <TextInput
                    label={'Nome da anotação'}
                    setText={handleChange('name')}
                    onBlur={handleBlur('name')}
                    placeholder={'Nome da anotação'}
                    value={values.name}
                  />
                  <Select
                    options={translateOptions(categories)}
                    onChange={value => setFieldValue('category', value)}
                    modalTitle="Selecione uma categoria"
                    inputLabel="Selecionar Categoria"
                    inputPlaceHolder="Selecionar Categoria"
                    defaultValue={values.category}
                  />
                  <SelectMultiple
                    options={translateOptions(subjects)}
                    onChange={value => setFieldValue('subjects', value)}
                    modalTitle="Selecione vários assuntos"
                    inputPlaceHolder="Selecionar Assuntos"
                  />
                </>
              ) : null}
              {step === 2 ? (
                <TextEditor
                  onChange={content => setFieldValue('content', content)}
                  placeHolder="Escreva sua anotação aqui!"
                />
              ) : null}
            </View>
            {keyboardStatus === 'Keyboard Shown' ? null : (
              <View>
                <Button
                  label={step === 2 ? 'Criar Anotação' : 'Proximo'}
                  onPress={step === 2 ? handleSubmit : () => setStep(step + 1)}
                />
                <Text style={styles.stepIndicator}>{`${step} / 2`}</Text>
              </View>
            )}
          </View>
        )}
        initialValues={{
          name: '',
          category: get(route, 'params.categoryId', ''),
          subjects: [],
          content: '',
        }}
        onSubmit={values => submitAction(values)}
      />
    </DefaultContainerView>
  );
};

export default compose(
  withDatabase,
  withObservables([], ({database}: any) => {
    return {
      categories: database.get('categories').query(),
      subjects: database.get('subjects').query(),
    };
  }),
)(Create);
