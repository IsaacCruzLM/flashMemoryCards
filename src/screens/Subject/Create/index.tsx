import React from 'react';
import {View} from 'react-native';
import withObservables from '@nozbe/with-observables';
import {withDatabase} from '@nozbe/watermelondb/DatabaseProvider';
import {compose} from 'recompose';
import get from 'lodash/get';
import {of as $of} from 'rxjs';

import NavigationService from '../../../navigation/NavigationService';
import WmdbUtils from '../../../databases/utils';
import ErrorHandlers from '../../../utils/errorHandlers';

import DefaultContainerView from '../../../components/DefaultContainerView';
import TextInput from '../../../components/TextInput';
import Form from '../../../components/Form';
import ColorPicker from '../../../components/ColorPicker';
import Button from '../../../components/Button';

import styles from './styles';
import {CreateProps, CreateFormProps, formValues} from './types';

const Create: React.FunctionComponent<any> = ({
  route,
  subject,
}: CreateProps) => {
  const isEdit = get(route, 'params.isEdit', false);
  const subjectId = get(route, 'params.subjectId', '');

  const cancelAction = (resetForm: () => void) => {
    resetForm();
    NavigationService.navigate('Subjects');
  };

  const submitAction = async (values: formValues | Object) => {
    isEdit
      ? await WmdbUtils.updateItemInWMDB('subjects', values, subjectId)
      : await WmdbUtils.insertItemInWMDB('subjects', {
          ...values,
          createdAt: new Date(),
        });
    NavigationService.navigate('Subjects');
  };

  const validate = (values: formValues) => {
    const errors = {} as formValues;

    if (!values.name) {
      errors.name = 'Campo "Nome" obrigatório';
    }
    if (!values.color) {
      errors.color = 'Campo "Cor" obrigatório';
    }

    return errors;
  };

  return (
    <DefaultContainerView>
      <Form
        validate={validate}
        form={({
          handleChange,
          handleBlur,
          setFieldValue,
          resetForm,
          handleSubmit,
          values,
          errors,
          touched,
        }: CreateFormProps) => (
          <View style={styles.formContainer}>
            <TextInput
              label={'Nome do assunto'}
              setText={handleChange('name')}
              onBlur={handleBlur('name')}
              placeholder={'Nome do assunto'}
              value={values.name}
              error={ErrorHandlers.showError(
                errors,
                touched,
                'name' as keyof Object,
              )}
              errorLabel={ErrorHandlers.getErrorLabel(
                errors,
                'name' as keyof Object,
              )}
            />
            <ColorPicker
              value={values.color}
              iconName={''}
              onChangeColor={color => setFieldValue('color', color)}
              error={ErrorHandlers.showError(
                errors,
                touched,
                'color' as keyof Object,
              )}
              errorLabel={ErrorHandlers.getErrorLabel(
                errors,
                'color' as keyof Object,
              )}
            />
            <Button
              label={`${isEdit ? 'Editar' : 'Criar'}  Categoria`}
              onPress={handleSubmit}
              style={styles.createButton}
            />
            <Button
              label="Cancelar"
              modeParam="outlined"
              onPress={() => cancelAction(resetForm)}
            />
          </View>
        )}
        initialValues={
          isEdit
            ? {name: subject.name, color: subject.color}
            : {name: '', color: ''}
        }
        onSubmit={values => submitAction(values)}
      />
    </DefaultContainerView>
  );
};

export default compose(
  withDatabase,
  withObservables([], ({database, route}: any) => {
    const isEdit = get(route, 'params.isEdit', false);
    const subjectId = get(route, 'params.subjectId', '');

    return {
      subject: isEdit
        ? database.get('subjects').findAndObserve(subjectId)
        : $of(null),
    };
  }),
)(Create);
