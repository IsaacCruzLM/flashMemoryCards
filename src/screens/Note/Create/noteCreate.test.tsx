import React from 'react';
import renderer from 'react-test-renderer';
import Create from './index';

jest.mock('@react-navigation/elements', () => {
  return {
    __esModule: true,
    useHeaderHeight: jest.fn().mockReturnValue(50), // Retorna a altura mockada
  };
});

describe('Note Create Screen', () => {
  let tree: renderer.ReactTestRenderer;
  beforeEach(() => {
    jest.clearAllMocks();

    tree = renderer.create(<Create />);
  });

  afterEach(() => {
    jest.clearAllMocks();
    jest.useRealTimers();
  });

  it('should render correctly', () => {
    expect(tree.toJSON()).toBeTruthy();
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('render all text inputs correctly', () => {
    const root = tree.root;

    const textInputs = root.findAllByProps({testID: 'text-input-outlined'});
    const textLabels = root.findAllByProps({
      testID: 'text-input-outlined-label-active',
    });
    const textPlaceholders = root.findAllByProps({
      testID: 'text-input-outlined-label-inactive',
    });

    expect(textInputs).toHaveLength(4);
    expect(textLabels).toHaveLength(14);
    expect(textPlaceholders).toHaveLength(14);
  });

  it('render categories select correctly', () => {
    const root = tree.root;

    const textInputLabel = root.find(
      node =>
        (node.type as String) === 'Text' &&
        node.props.children === 'Selecione uma categoria',
    );

    expect(textInputLabel).toBeTruthy();

    const selectMultipleInput = root.findByProps({
      testID: 'mockBottomSheetFlatList Selecione uma categoria',
    });

    expect(selectMultipleInput).toBeTruthy();
    expect(selectMultipleInput.props.data).toHaveLength(2);
  });

  it('render subjects select correctly', () => {
    const root = tree.root;

    const textInputLabel = root.findAll(
      node =>
        (node.type as String) === 'Text' &&
        node.props.children === 'Selecionar Assuntos',
    );

    expect(textInputLabel).toHaveLength(2);

    const selectMultipleInput = root.findByProps({
      testID: 'mockBottomSheetFlatList Selecione vÃ¡rios assuntos',
    });

    expect(selectMultipleInput).toBeTruthy();
    expect(selectMultipleInput.props.data).toHaveLength(2);
  });

  it('render action button correctly', () => {
    const root = tree.root;

    const buttonToCancel = root.findByProps({
      testID: 'ButtonTestID Proximo',
    });

    expect(buttonToCancel).toBeTruthy();
  });
});
