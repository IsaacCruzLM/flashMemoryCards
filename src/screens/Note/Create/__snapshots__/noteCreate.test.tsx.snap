// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note Create Screen should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-between",
        "paddingHorizontal": 20,
        "paddingVertical": 24,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "width": "100%",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <View>
      <View
        style={
          Object {
            "height": 36,
            "marginBottom": 48,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "#B00020",
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Array [
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 6,
                },
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "rgba(0, 0, 0, 0.54)",
                  "borderRadius": 4,
                  "borderWidth": 1,
                },
                undefined,
              ]
            }
            testID="text-input-outline"
          />
          <View>
            <View
              style={
                Array [
                  Object {
                    "paddingBottom": 0,
                  },
                  Object {
                    "minHeight": 64,
                    "paddingTop": 8,
                  },
                ]
              }
            >
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                style={
                  Object {
                    "backgroundColor": "#FFFFFF",
                    "color": "transparent",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 8,
                    "letterSpacing": 0.25,
                    "lineHeight": undefined,
                    "opacity": 0,
                    "paddingHorizontal": 0,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 37,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                      Object {
                        "scaleY": 1,
                      },
                    ],
                    "width": -14,
                    "writingDirection": "ltr",
                  }
                }
              >
                Nome da anotação
              </Text>
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "#22272E",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "letterSpacing": 0.4,
                    "lineHeight": undefined,
                    "opacity": 0,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 36,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
                testID="text-input-outlined-label-active"
              >
                Nome da anotação
              </Text>
              <Text
                collapsable={false}
                maxFontSizeMultiplier={1.5}
                numberOfLines={1}
                style={
                  Object {
                    "color": "rgba(0, 0, 0, 0.54)",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": undefined,
                    "left": 0,
                    "letterSpacing": 0.4,
                    "lineHeight": undefined,
                    "opacity": 0,
                    "paddingHorizontal": 14,
                    "position": "absolute",
                    "textAlign": "left",
                    "top": 36,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateY": 0,
                      },
                      Object {
                        "scale": 1,
                      },
                    ],
                    "writingDirection": "ltr",
                  }
                }
                testID="text-input-outlined-label-inactive"
              >
                Nome da anotação
              </Text>
              <TextInput
                editable={true}
                maxFontSizeMultiplier={1.5}
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder=" "
                placeholderTextColor="rgba(0, 0, 0, 0.54)"
                selectionColor="#22272E"
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "margin": 0,
                      "zIndex": 1,
                    },
                    Object {
                      "height": 56,
                    },
                    Object {
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "lineHeight": undefined,
                      "paddingHorizontal": 14,
                      "textAlign": "left",
                      "textAlignVertical": "center",
                    },
                    false,
                    Array [
                      Object {},
                    ],
                  ]
                }
                testID="text-input-outlined"
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "justifyContent": "center",
            "paddingBottom": 8,
          }
        }
      >
        <View
          pointerEvents="none"
          style={Object {}}
        >
          <View
            style={
              Object {
                "height": 36,
                "marginBottom": 48,
                "width": "100%",
              }
            }
          >
            <View
              style={
                Object {
                  "borderColor": "#B00020",
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Array [
                    Object {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 6,
                    },
                    Object {
                      "backgroundColor": "#FFFFFF",
                      "borderColor": "rgba(0, 0, 0, 0.54)",
                      "borderRadius": 4,
                      "borderWidth": 1,
                    },
                    undefined,
                  ]
                }
                testID="text-input-outline"
              />
              <View>
                <View
                  style={
                    Array [
                      Object {
                        "paddingBottom": 0,
                      },
                      Object {
                        "minHeight": 64,
                        "paddingTop": 8,
                      },
                    ]
                  }
                >
                  <Text
                    collapsable={false}
                    maxFontSizeMultiplier={1.5}
                    numberOfLines={1}
                    style={
                      Object {
                        "backgroundColor": "#FFFFFF",
                        "color": "transparent",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 8,
                        "letterSpacing": 0.25,
                        "lineHeight": undefined,
                        "opacity": 0,
                        "paddingHorizontal": 0,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 37,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 1,
                          },
                          Object {
                            "scaleY": 1,
                          },
                        ],
                        "width": -14,
                        "writingDirection": "ltr",
                      }
                    }
                  >
                    Selecionar Categoria
                  </Text>
                  <Text
                    collapsable={false}
                    maxFontSizeMultiplier={1.5}
                    numberOfLines={1}
                    onLayout={[Function]}
                    style={
                      Object {
                        "color": "#22272E",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 0,
                        "letterSpacing": 0.4,
                        "lineHeight": undefined,
                        "opacity": 0,
                        "paddingHorizontal": 14,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 36,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                    testID="text-input-outlined-label-active"
                  >
                    Selecionar Categoria
                  </Text>
                  <Text
                    collapsable={false}
                    maxFontSizeMultiplier={1.5}
                    numberOfLines={1}
                    style={
                      Object {
                        "color": "rgba(0, 0, 0, 0.54)",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "left": 0,
                        "letterSpacing": 0.4,
                        "lineHeight": undefined,
                        "opacity": 0,
                        "paddingHorizontal": 14,
                        "position": "absolute",
                        "textAlign": "left",
                        "top": 36,
                        "transform": Array [
                          Object {
                            "translateX": 0,
                          },
                          Object {
                            "translateY": 0,
                          },
                          Object {
                            "scale": 1,
                          },
                        ],
                        "writingDirection": "ltr",
                      }
                    }
                    testID="text-input-outlined-label-inactive"
                  >
                    Selecionar Categoria
                  </Text>
                  <TextInput
                    editable={true}
                    maxFontSizeMultiplier={1.5}
                    multiline={false}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder=" "
                    placeholderTextColor="rgba(0, 0, 0, 0.54)"
                    selectionColor="#22272E"
                    style={
                      Array [
                        Object {
                          "flexGrow": 1,
                          "margin": 0,
                          "zIndex": 1,
                        },
                        Object {
                          "height": 56,
                        },
                        Object {
                          "paddingBottom": 0,
                          "paddingTop": 0,
                        },
                        Object {
                          "color": "#000000",
                          "fontFamily": "System",
                          "fontSize": 16,
                          "fontWeight": undefined,
                          "lineHeight": undefined,
                          "paddingHorizontal": 14,
                          "textAlign": "left",
                          "textAlignVertical": "center",
                        },
                        false,
                        Array [
                          Object {},
                        ],
                      ]
                    }
                    testID="text-input-outlined"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          testID="mockBottomSheetModal"
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingHorizontal": 16,
                  "width": "100%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#22272E",
                    "fontFamily": "System",
                    "fontSize": 20,
                    "fontWeight": "600",
                    "textAlign": "left",
                  }
                }
              >
                Selecione uma categoria
              </Text>
              <View
                accessibilityComponentType="button"
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": false,
                  }
                }
                accessibilityTraits="button"
                accessible={true}
                centered={true}
                collapsable={false}
                focusable={true}
                hitSlop={
                  Object {
                    "bottom": 6,
                    "left": 6,
                    "right": 6,
                    "top": 6,
                  }
                }
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    false,
                    Object {
                      "alignItems": "center",
                      "flexGrow": 1,
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Icon
                  accessibilityElementsHidden={true}
                  color="#FFFFFF"
                  importantForAccessibility="no-hide-descendants"
                  name="close"
                  pointerEvents="none"
                  selectable={false}
                  size={20}
                  style={
                    Array [
                      Object {
                        "lineHeight": 20,
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ]
                  }
                />
              </View>
            </View>
            <View
              data={
                Array [
                  Object {
                    "iconColor": "#d11cd5",
                    "iconName": undefined,
                    "label": "Categoria 1",
                    "value": "category_1",
                  },
                  Object {
                    "iconColor": "#f2f2f2",
                    "iconName": undefined,
                    "label": "Categoria 2",
                    "value": "category_2",
                  },
                ]
              }
              keyExtractor={[Function]}
              renderItem={[Function]}
              testID="mockBottomSheetFlatList Selecione uma categoria"
            />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "justifyContent": "center",
            "marginBottom": 16,
          }
        }
      >
        <View
          pointerEvents="none"
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                  "minHeight": 58,
                  "paddingHorizontal": 8,
                  "paddingVertical": 8,
                  "width": "100%",
                },
                Object {
                  "borderColor": "gray",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 4,
                  "left": 14,
                  "position": "absolute",
                  "top": -10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 12,
                  }
                }
              >
                Selecionar Assuntos
              </Text>
            </View>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "System",
                    "fontSize": 16,
                    "fontWeight": "600",
                    "height": 40,
                    "marginLeft": 5.6,
                    "textAlign": "left",
                    "textAlignVertical": "center",
                  },
                  Object {
                    "color": "#162133",
                  },
                ]
              }
            >
              Selecionar Assuntos
            </Text>
          </View>
        </View>
        <View
          testID="mockBottomSheetModal"
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingHorizontal": 16,
                  "width": "100%",
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#22272E",
                    "fontFamily": "System",
                    "fontSize": 20,
                    "fontWeight": "600",
                    "textAlign": "left",
                  }
                }
              >
                Selecione vários assuntos
              </Text>
              <View
                accessibilityComponentType="button"
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": false,
                  }
                }
                accessibilityTraits="button"
                accessible={true}
                centered={true}
                collapsable={false}
                focusable={true}
                hitSlop={
                  Object {
                    "bottom": 6,
                    "left": 6,
                    "right": 6,
                    "top": 6,
                  }
                }
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    false,
                    Object {
                      "alignItems": "center",
                      "flexGrow": 1,
                      "justifyContent": "center",
                    },
                  ]
                }
              >
                <Icon
                  accessibilityElementsHidden={true}
                  color="#FFFFFF"
                  importantForAccessibility="no-hide-descendants"
                  name="close"
                  pointerEvents="none"
                  selectable={false}
                  size={20}
                  style={
                    Array [
                      Object {
                        "lineHeight": 20,
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ]
                  }
                />
              </View>
            </View>
            <View
              data={
                Array [
                  Object {
                    "iconColor": "#ffffff",
                    "iconName": undefined,
                    "label": "Assunto 1",
                    "value": "subject_1",
                  },
                  Object {
                    "iconColor": "#d11cd5",
                    "iconName": undefined,
                    "label": "Assunto 2",
                    "value": "subject_2",
                  },
                ]
              }
              keyExtractor={[Function]}
              renderItem={[Function]}
              testID="mockBottomSheetFlatList Selecione vários assuntos"
            />
          </View>
        </View>
      </View>
    </View>
    <View>
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            false,
            Object {
              "borderRadius": 20,
            },
          ]
        }
        testID="ButtonTestID Proximo"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            selectable={false}
            style={
              Array [
                Object {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                Object {
                  "textAlign": "left",
                },
                Object {
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  false,
                  Object {
                    "marginHorizontal": 24,
                    "marginVertical": 10,
                  },
                  undefined,
                  false,
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "500",
                    "letterSpacing": 0.1,
                    "lineHeight": 20,
                  },
                  undefined,
                ],
              ]
            }
          >
            Proximo
          </Text>
        </View>
      </View>
      <Text
        style={
          Object {
            "color": "#000000B3",
            "fontFamily": "System",
            "fontSize": 16,
            "fontWeight": "700",
            "textAlign": "center",
          }
        }
      >
        1 / 2
      </Text>
    </View>
  </View>
</View>
`;
